services/api/tests/conftest.py

```diff
@@ -1,24 +1,41 @@
 import os
 from pathlib import Path
+from urllib.parse import urlparse
 
+import numpy as np
 import pytest
 import xarray as xr
+from s3fs import S3FileSystem, S3Map
 
 from unified_graphics import create_app
 
 
-@pytest.fixture(autouse=True)
-def clear_aws_credentials():
-    """Mocked AWS Credentials for moto
+def pytest_addoption(parser):
+    parser.addoption(
+        "--runaws",
+        action="store_true",
+        default=False,
+        help="run tests that require authentication to AWS",
+    )
 
-    Run this before every test to make sure we don't push to an actual S3
-    instance when running tests.
-    """
-    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
-    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
-    os.environ["AWS_SECURITY_TOKEN"] = "testing"
-    os.environ["AWS_SESSION_TOKEN"] = "testing"
-    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers", "aws: mark test as requiring an authenticated connection to AWS"
+    )
+
+
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--runaws"):
+        # If --runaws is set, there's no need to modify any of the tests
+        # to mark them skipped.
+        return
+
+    # Mark any test with the aws mark as skipped because --runaws is off
+    skip_aws = pytest.mark.skip(reason="use --runaws to run")
+    for item in items:
+        if "aws" in item.keywords:
+            item.add_marker(skip_aws)
```

Comment:
This is a nice way to automatically disable tests.

Reply:
Straight out of the pytest docs. I always forget where it is, but I know it’s in there, so I just have to keep digging until I find it.